### PR TITLE
Add methylation resource for chrX

### PR DESCRIPTION
**Pull request by klaricch**

### CODE REVIEW

**Comment by klaricch** [comment]
I don't have write access, so I can't actually put those files at the specified paths

File: gnomad/resources/grch38/reference_data.py [hunk 1]
```diff
@@ -322,6 +322,15 @@ def _import_ensembl_interval(path) -> hl.Table:
     },
 )
 
+# Methylation scores range from 0-12.
```
Comment (jkgoodrich) [inline gnomad/resources/grch38/reference_data.py:325]: Are these from the same `(https://www.biorxiv.org/content/10.1101/2022.03.20.485034v2.full).` or if not is there a comment we can add indicating where they came from?

File: gnomad/resources/grch38/reference_data.py [hunk 2]
```diff
@@ -322,6 +322,15 @@ def _import_ensembl_interval(path) -> hl.Table:
...
+methylation_sites_chrx = GnomadPublicTableResource(
+    path="gs://gcp-public-data--gnomad/resources/grch38/methylation_sites/methylation_chrX.ht",
```
Comment (jkgoodrich) [inline gnomad/resources/grch38/reference_data.py:327]:
```suggestion
    path="gs://gnomad-public-requester-pays/resources/grch38/methylation_sites/methylation_chrX.ht",
```

File: gnomad/resources/grch38/reference_data.py [hunk 3]
```diff
@@ -322,6 +322,15 @@ def _import_ensembl_interval(path) -> hl.Table:
...
+    import_func=_import_methylation_sites,
+    import_args={
+        "path": "gs://gcp-public-data--gnomad/resources/grch38/methylation_sites/methylation_chrX.bed",
```
Comment (jkgoodrich) [inline gnomad/resources/grch38/reference_data.py:330]:
```suggestion
        "path": "gs://gnomad-public-requester-pays/resources/grch38/methylation_sites/methylation_chrX.bed",
```

**Review by jkgoodrich — CHANGES_REQUESTED** [review]
If using `gs://gnomad-public-requester-pays/` it will work while it's only located in that folder too, and then it will still work as expected when that file is removed and it's only in the `gcp-public-data--gnomad` bucket

**Review by jkgoodrich — APPROVED** [review]
lgtm

**Pull request merged**